src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs

```diff
@@ -253,7 +253,11 @@ public Task WaitForMeasurementsAsync(int minCount, CancellationToken cancellatio
     [SuppressMessage("Resilience", "R9A061:The async method doesn't support cancellation", Justification = "Not relevant in this case")]
     public async Task WaitForMeasurementsAsync(int minCount, TimeSpan timeout)
     {
-        using var cancellationTokenSource = new CancellationTokenSource(timeout);
+#if NET8_0_OR_GREATER
+        using var cancellationTokenSource = new CancellationTokenSource(timeout, _timeProvider);
+#else
+        using var cancellationTokenSource = _timeProvider.CreateCancellationTokenSource(timeout);
+#endif
```

Comment:
I am wondering can we just use the extension method to avoid the ifdef? or Microsoft.Bcl.TimeProvider is not referenced at all when building for .NET 8.0 and up?

Reply:
Correct, the symbol doesn't exist when building for .NET 8.